ast/env.go

```diff
@@ -304,6 +304,116 @@ func (n *typeTreeNode) Put(path Ref, tpe types.Type) {
 	curr.value = tpe
 }
 
+// Insert inserts tpe at path in the tree, but also merges the value into any types.Object present along that path.
+// If an types.Object is inserted, any leafs already present further down the tree are merged into the inserted object.
+// path must be ground.
+func (n *typeTreeNode) Insert(path Ref, tpe types.Type) {
+	curr := n
+	for i, term := range path {
+		c, ok := curr.children.Get(term.Value)
+
+		var child *typeTreeNode
+		if !ok {
+			child = newTypeTree()
+			child.key = term.Value
```

Comment:
Do we need to check if this is an ast.Var? (Real question, I don't remember much about the type checker at this moment.)

Reply:
At this point, we know the passed `path` is ground.
This might change once I make the changes required for generic refs; but I'm not sure.